From 495ee9c0b61963fd5b5252030e2dee8dcf84d3cc Mon Sep 17 00:00:00 2001
From: Marco Sirabella <marco@sirabella.org>
Date: Fri, 26 Oct 2018 21:21:36 -0400
Subject: [PATCH 1/2] Allow intent shared subject or title to be mms subject

Fix some styling issues

Thanks to LuK1337 on the irc channel for helpful pointers

Change-Id: Ibb5eba030320467774cdac1d502ec36c98abcb43
---
 .../messaging/datamodel/data/MessageData.java      |  4 +++-
 .../ui/conversationlist/ShareIntentActivity.java   | 14 +++++++++++---
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/src/com/android/messaging/datamodel/data/MessageData.java b/src/com/android/messaging/datamodel/data/MessageData.java
index a3698a9..8229a9d 100644
--- a/src/com/android/messaging/datamodel/data/MessageData.java
+++ b/src/com/android/messaging/datamodel/data/MessageData.java
@@ -297,9 +297,11 @@ public class MessageData implements Parcelable {
     /**
      * Create a message not yet associated with a particular conversation
      */
-    public static MessageData createSharedMessage(final String messageText) {
+    public static MessageData createSharedMessage(final String messageText,
+            final String subjectText) {
         final MessageData message = new MessageData();
         message.mStatus = BUGLE_STATUS_OUTGOING_DRAFT;
+        message.mMmsSubject = subjectText;
         if (!TextUtils.isEmpty(messageText)) {
             message.mParts.add(MessagePartData.createTextMessagePart(messageText));
         }
diff --git a/src/com/android/messaging/ui/conversationlist/ShareIntentActivity.java b/src/com/android/messaging/ui/conversationlist/ShareIntentActivity.java
index 6d51de2..be94fa1 100644
--- a/src/com/android/messaging/ui/conversationlist/ShareIntentActivity.java
+++ b/src/com/android/messaging/ui/conversationlist/ShareIntentActivity.java
@@ -77,6 +77,14 @@ public class ShareIntentActivity extends BaseBugleActivity implements
     public void onAttachFragment(final Fragment fragment) {
         final Intent intent = getIntent();
         final String action = intent.getAction();
+
+        String sharedSubject = null;
+        if (intent.hasExtra(Intent.EXTRA_SUBJECT)) {
+            sharedSubject = intent.getStringExtra(Intent.EXTRA_SUBJECT);
+        } else if (intent.hasExtra(Intent.EXTRA_TITLE)) {
+            sharedSubject = intent.getStringExtra(Intent.EXTRA_TITLE);
+        }
+
         if (Intent.ACTION_SEND.equals(action)) {
             final Uri contentUri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
             if (UriUtil.isFileUri(contentUri)) {
@@ -94,7 +102,7 @@ public class ShareIntentActivity extends BaseBugleActivity implements
             if (ContentType.TEXT_PLAIN.equals(contentType)) {
                 final String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
                 if (sharedText != null) {
-                    mDraftMessage = MessageData.createSharedMessage(sharedText);
+                    mDraftMessage = MessageData.createSharedMessage(sharedText, sharedSubject);
                 } else {
                     mDraftMessage = null;
                 }
@@ -103,7 +111,7 @@ public class ShareIntentActivity extends BaseBugleActivity implements
                     ContentType.isAudioType(contentType) ||
                     ContentType.isVideoType(contentType)) {
                 if (contentUri != null) {
-                    mDraftMessage = MessageData.createSharedMessage(null);
+                    mDraftMessage = MessageData.createSharedMessage(null, sharedSubject);
                     addSharedImagePartToDraft(contentType, contentUri);
                 } else {
                     mDraftMessage = null;
@@ -123,7 +131,7 @@ public class ShareIntentActivity extends BaseBugleActivity implements
                 final ArrayList<Uri> imageUris = intent.getParcelableArrayListExtra(
                         Intent.EXTRA_STREAM);
                 if (imageUris != null && imageUris.size() > 0) {
-                    mDraftMessage = MessageData.createSharedMessage(null);
+                    mDraftMessage = MessageData.createSharedMessage(null, sharedSubject);
                     for (final Uri imageUri : imageUris) {
                         if (UriUtil.isFileUri(imageUri)) {
                             LogUtil.i(
-- 
2.17.1

